packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -160,21 +161,93 @@ export default createRule<Options, MessageIds>({
     function getEsNodesFromViolatingNode(
       violatingNode: ParameterOrPropertyDeclaration,
     ): { esNode: TSESTree.Node; nameNode: TSESTree.Node } {
-      if (
-        ts.isParameterPropertyDeclaration(violatingNode, violatingNode.parent)
-      ) {
-        return {
-          esNode: services.tsNodeToESTreeNodeMap.get(violatingNode.name),
-          nameNode: services.tsNodeToESTreeNodeMap.get(violatingNode.name),
-        };
-      }
-
       return {
         esNode: services.tsNodeToESTreeNodeMap.get(violatingNode),
         nameNode: services.tsNodeToESTreeNodeMap.get(violatingNode.name),
       };
     }
 
+    /**
+     * For missing readonly modifiers, we want to report any keywords
+     * out in front of the key, and the key itself, but not anything afterwards,
+     * i.e. parens, type annotations, method bodies, or `?`.
+     */
+    function getReportLoc(
```

Comment:
extract and reuse utilities from https://github.com/typescript-eslint/typescript-eslint/pull/8869